src/vector_of_vectors.jl

```diff
@@ -0,0 +1,96 @@
+# Data structure that behaves like a `Vector{Vector}`, but uses a contiguous memory layout.
+# Similar to `VectorOfVectors` of ArraysOfArrays.jl, but allows to resize the inner vectors.
+struct DynamicVectorOfVectors{T, ARRAY2D, ARRAY1D} <: AbstractVector{Array{T, 1}}
```

Comment:
I'm open for better naming suggestions.

Reply:
NestedDynamicVector?

Reply:
ContiguousVectorMatrix

Reply:
I don't think any of these are more descriptive. I have to close the conversation to merge, but I'm still open for suggestions.